server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessageListMethodStepdefs.java

```diff
@@ -72,6 +74,42 @@ public void getMessageList(String username, List<String> mailboxes, String flag)
             flag));
     }
 
+    @When("^\"([^\"]*)\" asks for message list in (?:mailboxes|mailbox) \"([^\"]*)\"$")
+    public void getMessageList(String username, List<String> mailboxes) throws Exception {
+        String mailboxIds = Joiner.on("\",\"")
```

Comment:
The way to generate this _mailboxIds_ string seems duplicated everywhere.

Reply:
Done, and yes it was messy and everywhere